cannot install

Hi I haven't had any success with installing the program with `conda` or `docker` and the `installUbuntu.sh` script is also giving me issues.

> cmake  -DTBB_DIR=${PWD}/oneTBB-2019_U9  -DCMAKE_PREFIX_PATH=${PWD}/oneTBB-2019_U9/cmake ..
> -- 0.5.2
> -- Using Protocol Buffers 3.20.0.1
> -- Building Intel TBB: /usr/bin/make -j64 tbb_build_prefix=tbb_cmake_build_subdir tbb_build_dir=/space/s1/lenore/software/usher/build/tbb_cmake_build compiler=gcc tbb_cpf=1
> -- Boost version: 1.65.1
> -- Found the following Boost libraries:
> --   program_options
> --   iostreams
> --   filesystem
> --   date_time
> --   regex
> --   system
> CMake Error at CMakeLists.txt:343 (TARGET_LINK_LIBRARIES):
>   Cannot specify link libraries for target "output_final_protobuf" which is
>   not built by this project.
> 
> 
> -- Configuring incomplete, errors occurred!

Hi, I tried your fix but now I'm getting a different error

> cmake  -DTBB_DIR=/space/s1/lenore/software/usher/build/oneTBB-2019_U9  -DCMAKE_PREFIX_PATH=/space/s1/lenore/software/usher/build/oneTBB-2019_U9/cmake ..
> -- 0.5.2
> -- Using Protocol Buffers 3.20.0.1
> -- Building Intel TBB: /usr/bin/make -j64 tbb_build_prefix=tbb_cmake_build_subdir tbb_build_dir=/space/s1/lenore/software/usher/build/tbb_cmake_build compiler=gcc tbb_cpf=1
> -- Boost version: 1.65.1
> -- Found the following Boost libraries:
> --   program_options
> --   iostreams
> --   filesystem
> --   date_time
> --   regex
> --   system
> CMake Error at CMakeLists.txt:340 (TARGET_LINK_LIBRARIES):
>   Cannot specify link libraries for target "transposed_vcf_to_fa" which is
>   not built by this project.
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "/space/s1/lenore/software/usher/build/CMakeFiles/CMakeOutput.log".
> See also "/space/s1/lenore/software/usher/build/CMakeFiles/CMakeError.log".

Ok I added these lines when Protobuf is defined

>     add_executable(transposed_vcf_to_vcf
>         src/matOptimize/transpose_vcf/transposed_vcf_to_vcf.cpp
>         src/matOptimize/mutation_annotated_tree_nuc_util.cpp
>         )
>         add_executable(transposed_vcf_to_fa
>         src/matOptimize/mutation_annotated_tree_nuc_util.cpp
>         src/matOptimize/transpose_vcf/transposed_vcf_to_fa.cpp
>         )
>         add_executable(transposed_vcf_print_name
>         src/matOptimize/transpose_vcf/transposed_vcf_print_name.cpp
>         src/matOptimize/mutation_annotated_tree_nuc_util.cpp
>         )

But now I'm getting yet another error

> cmake  -DTBB_DIR=/space/s1/lenore/software/usher/build/oneTBB-2019_U9  -DCMAKE_PREFIX_PATH=/space/s1/lenore/software/usher/build/oneTBB-2019_U9/cmake ..
> -- 0.5.2
> -- Using Protocol Buffers 3.20.0.1
> -- Building Intel TBB: /usr/bin/make -j64 tbb_build_prefix=tbb_cmake_build_subdir tbb_build_dir=/space/s1/lenore/software/usher/build/tbb_cmake_build compiler=gcc tbb_cpf=1
> -- Boost version: 1.65.1
> -- Found the following Boost libraries:
> --   program_options
> --   iostreams
> --   filesystem
> --   date_time
> --   regex
> --   system
> -- Configuring done
> CMake Error at CMakeLists.txt:85 (add_executable):
>   Cannot find source file:
> 
>     src/matOptimize/detailed_mutations_load_store.cpp
> 
>   Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
>   .hxx .in .txx
> 
> 
> CMake Error: CMake can not determine linker language for target: matOptimize
> CMake Error: Cannot determine link language for target "matOptimize".
> -- Generating done
> -- Build files have been written to: /space/s1/lenore/software/usher/build

Also I tried to compile just the `matUtils extract` function and got this error

> g++ -o extract extract.cpp
> In file included from ../usher_graph.hpp:3:0,
>                  from common.hpp:11,
>                  from extract.hpp:1,
>                  from extract.cpp:1:
> ../mutation_annotated_tree.hpp:21:10: fatal error: parsimony.pb.h: No such file or directory
>  #include "parsimony.pb.h"
>           ^~~~~~~~~~~~~~~~
> compilation terminated.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,6 +128,14 @@ if(DEFINED Protobuf_PATH)
         src/matOptimize/transpose_vcf/transposed_vcf_to_vcf.cpp
         src/matOptimize/mutation_annotated_tree_nuc_util.cpp
         )
+    add_executable(transposed_vcf_to_fa
+    src/matOptimize/mutation_annotated_tree_nuc_util.cpp
+    src/matOptimize/transpose_vcf/transposed_vcf_to_fa.cpp
+    )
+    add_executable(transposed_vcf_print_name
+    src/matOptimize/transpose_vcf/transposed_vcf_print_name.cpp
+    src/matOptimize/mutation_annotated_tree_nuc_util.cpp
+    )
     if(USHER_SERVER)
         add_executable(usher_server
             src/mutation_annotated_tree.cpp
@@ -173,7 +181,7 @@ if(DEFINED Protobuf_PATH)
     protobuf_generate(
         LANGUAGE cpp
         TARGET output_final_protobuf
-        PROTOS mutation_detailed.proto)
+        PROTOS parsimony.proto)
     protobuf_generate(
         LANGUAGE cpp
         TARGET output_final_protobuf

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,16 @@ if(DEFINED Protobuf_PATH)
         src/usher_mapper.cpp
         )
 
-
+    add_executable(output_final_protobuf
+        src/matOptimize/mutation_annotated_tree.cpp
+        src/matOptimize/mutation_annotated_tree_node.cpp
+        src/matOptimize/detailed_mutations_load.cpp
+        src/matOptimize/mutation_annotated_tree_load_store.cpp
+        src/matOptimize/main_helper.cpp
+        src/matOptimize/Mutation_Collection.cpp
+        src/matOptimize/mutation_annotated_tree_nuc_util.cpp
+        src/matOptimize/output_final_protobuf.cpp
+    )
     add_executable(transpose_vcf
         src/matOptimize/transpose_vcf/transpose_vcf_encode.cpp
         src/matOptimize/mutation_annotated_tree_nuc_util.cpp
@@ -161,7 +170,14 @@ if(DEFINED Protobuf_PATH)
         LANGUAGE cpp
         TARGET matOptimize
         PROTOS mutation_detailed.proto)
-   
+    protobuf_generate(
+        LANGUAGE cpp
+        TARGET output_final_protobuf
+        PROTOS mutation_detailed.proto)
+    protobuf_generate(
+        LANGUAGE cpp
+        TARGET output_final_protobuf
+        PROTOS mutation_detailed.proto)
     protobuf_generate(
         LANGUAGE cpp
         TARGET matUtils
@@ -288,28 +304,6 @@ else()
         src/matOptimize/transpose_vcf/transposed_vcf_print_name.cpp
         src/matOptimize/mutation_annotated_tree_nuc_util.cpp
         )
-        #[[add_executable(serialize_test
-        src/matOptimize/mutation_annotated_tree.cpp
-        src/matOptimize/mutation_annotated_tree_node.cpp
-        src/matOptimize/serializer_test.cpp
-        src/matOptimize/detailed_mutations_store.cpp
-        src/matOptimize/mutation_annotated_tree_load_store.cpp
-        src/matOptimize/detailed_mutations_load.cpp
-        src/matOptimize/mutation_annotated_tree_nuc_util.cpp
-        src/matOptimize/import_vcf_fast.cpp
-        src/matOptimize/condense.cpp
-        src/matOptimize/VCF_load_tree.cpp
-        src/matOptimize/main_load_tree.cpp
-        src/matOptimize/main_helper.cpp
-        src/matOptimize/Mutation_Collection.cpp
-        src/matOptimize/Fitch_Sankoff.cpp
-        src/matOptimize/check_samples.cpp
-        ${PROTO_SRCS}
-        ${PROTO_HDRS}
-        ${patch_tree}
-        ${DETAILED_MUTATIONS_PROTO_SRCS}
-        ${DETAILED_MUTATIONS_PROTO_HDRS}
-        )]]
     if(USHER_SERVER)
         add_executable(usher_server
             src/mutation_annotated_tree.cpp
@@ -345,7 +339,6 @@ TARGET_LINK_LIBRARIES(transpose_vcf PRIVATE stdc++  ${Boost_LIBRARIES} ${TBB_IMP
 TARGET_LINK_LIBRARIES(transposed_vcf_to_vcf PRIVATE stdc++  ${Boost_LIBRARIES} ${TBB_IMPORTED_TARGETS} ${Protobuf_LIBRARIES} ZLIB::ZLIB) # OpenMP::OpenMP_CXX)
 TARGET_LINK_LIBRARIES(transposed_vcf_to_fa PRIVATE stdc++  ${Boost_LIBRARIES} ${TBB_IMPORTED_TARGETS} ${Protobuf_LIBRARIES} ZLIB::ZLIB) # OpenMP::OpenMP_CXX)
 TARGET_LINK_LIBRARIES(transposed_vcf_print_name PRIVATE stdc++  ${Boost_LIBRARIES} ${TBB_IMPORTED_TARGETS} ${Protobuf_LIBRARIES} ZLIB::ZLIB) # OpenMP::OpenMP_CXX)
-#TARGET_LINK_LIBRARIES(serialize_test PRIVATE stdc++  ${Boost_LIBRARIES} ${TBB_IMPORTED_TARGETS} ${Protobuf_LIBRARIES} ZLIB::ZLIB) # OpenMP::OpenMP_CXX)
 
 TARGET_LINK_LIBRARIES(usher PRIVATE stdc++  ${Boost_LIBRARIES} ${TBB_IMPORTED_TARGETS} ${Protobuf_LIBRARIES} ZLIB::ZLIB) # OpenMP::OpenMP_CXX)
 target_include_directories(usher PUBLIC "${PROJECT_BINARY_DIR}")

